core/tests/protractor_utils/action.js

```diff
@@ -32,6 +32,18 @@ var click = async function(elementName, clickableElement) {
   await clickableElement.click();
 };
 
+var count = async function(elementsName, elements) {
+  await waitFor.visibilityOf(elements, `${elementsName} is not visible.`);
+
+  return await textElement.count();
+};
```

Comment:
What if the count is zero because there are no elements visible?

Reply:
`waitFor.visibilityOf(` will fail in that case, right?

Reply:
I mean that if there are no elements visible, this should return `0` instead of failing, right? Isn't that how `elem.count()` behaves?

Reply:
We will be using element.isPresent to check whether the elements are visible or not. We will use "count" mostly for non-zero elements. (not sure whether this is around your question.)

Reply:
We sometimes check that the count is 0:

https://github.com/oppia/oppia/blob/291b56286bc1bcf1c192cf435ab962dda7af134e/core/tests/protractor_utils/LearnerDashboardPage.js#L133

Also in other cases, we don't know what the expected count will be a priori, so we don't know whether to use `count()` or `isPresent`:

https://github.com/oppia/oppia/blob/291b56286bc1bcf1c192cf435ab962dda7af134e/core/tests/protractor_utils/SkillEditorPage.js#L191

I guess we could use an `if` statement to handle cases like these, but it'd be an easier migration if we can handle the case where the count is zero in the `action` method. Maybe with a try-except block? I think we also considered passing a failure handler when creating a promise?